src/Microsoft.Azure.WebJobs.Host/Config/ExtensionConfigContext.cs

```diff
@@ -63,12 +64,18 @@ public Uri GetWebhookHandler()
         /// <returns></returns>
         public FluentBindingRule<TAttribute> AddBindingRule<TAttribute>() where TAttribute : Attribute
         {
-            bool hasBindingAttr = typeof(TAttribute).GetCustomAttributes(typeof(BindingAttribute), false).Length > 0;
-            if (!hasBindingAttr)
+            var bindingAttrs = typeof(TAttribute).GetCustomAttributes(typeof(BindingAttribute), false);
+            if (!bindingAttrs.Any())
             {                
                 throw new InvalidOperationException($"Can't add a binding rule for '{typeof(TAttribute).Name}' since it is missing the a {typeof(BindingAttribute).Name}");
             }
 
+            var isTrigger = typeof(TAttribute).Name.EndsWith("TriggerAttribute", StringComparison.OrdinalIgnoreCase);
+            if (!isTrigger && ((BindingAttribute)bindingAttrs.First()).TriggerHandlesReturnValue)
+            {
+                throw new InvalidOperationException($"Only declare TriggerHandlesReturnValue property true for trigger bindings.");
```

Comment:
>TriggerHandlesReturnValue [](start = 67, length = 25)

Can this be nameof(BindingAttribute.TriggerHandlesReturnValue) ? That's a relatively new C# feature.